src/views/AddItem.jsx

```diff
@@ -1,3 +1,115 @@
-export function AddItem() {
-	return <p>Hello from the <code>/add-item</code> page!</p>
+import { useEffect, useState } from 'react';
+import { addItem } from '../api/firebase';
+
+const timeframeToDays = {
+	Soon: 7,
+	'Kind of Soon': 14,
+	'Not Soon': 30,
+};
+
+export function AddItem({ listId }) {
+	const [timeframe, setTimeframe] = useState('Soon');
+	const [itemName, setItemName] = useState('');
+	const [message, setMessage] = useState(null);
+
+	useEffect(() => {
+		const timer = setTimeout(() => {
+			setMessage(null);
+		}, 3000);
+
+		return () => clearTimeout(timer);
+	}, [message]);
+
+	const onTimeChange = (e) => setTimeframe(e.target.value);
+	const onItemChange = (e) => setItemName(e.target.value);
+
+	return (
+		<form
+			onSubmit={(event) => {
+				event.preventDefault();
```

Comment:
Consider making this separate function like `onTimeChange` and `onItemChange` above